internal/controller/pkg/revision/deployment.go

```diff
@@ -54,10 +54,18 @@ const (
 	essTLSCertDirEnvVar = "ESS_TLS_CERTS_DIR"
 	essCertsVolumeName  = "ess-client-certs"
 	essCertsDir         = "/ess/tls"
+
+	tlsServerCertDirEnvVar   = "TLS_SERVER_CERTS_DIR"
+	tlsServerCertsVolumeName = "tls-server-certs"
+	tlsServerCertsDir        = "/tls/server"
+
+	tlsClientCertDirEnvVar   = "TLS_CLIENT_CERTS_DIR"
+	tlsClientCertsVolumeName = "tls-client-certs"
+	tlsClientCertsDir        = "/tls/client"
 )
 
 //nolint:gocyclo // TODO(negz): Can this be refactored for less complexity (and fewer arguments?)
-func buildProviderDeployment(provider *pkgmetav1.Provider, revision v1.PackageRevision, cc *v1alpha1.ControllerConfig, namespace string, pullSecrets []corev1.LocalObjectReference) (*corev1.ServiceAccount, *appsv1.Deployment, *corev1.Service) {
+func buildProviderDeployment(provider *pkgmetav1.Provider, revision v1.PackageRevision, cc *v1alpha1.ControllerConfig, namespace string, pullSecrets []corev1.LocalObjectReference) (*corev1.ServiceAccount, *appsv1.Deployment, *corev1.Service, *corev1.Secret, *corev1.Secret) {
```

Comment:
can you add a connebt about the returned secrets or at least use named outputs to make it clearer what's returned?

Reply:
The amount of arguments and return values of this function concerns me. That said, I'm okay with it growing more temporarily since it will be refactored soon to implement https://github.com/crossplane/crossplane/blob/a888ea/design/one-pager-package-runtime-config.md